src/commands/classified.rs

```diff
@@ -235,53 +234,44 @@ impl ExternalCommand {
             arg_string.push_str(&arg);
         }
 
-        let mut process;
-
-        process = Exec::cmd(&self.name);
-
-        trace!(target: "nu::run::external", "command = {:?}", process);
+        trace!(target: "nu::run::external", "command = {:?}", self.name);
 
+        let mut process;
         if arg_string.contains("$it") {
-            let mut first = true;
-
-            for i in &inputs {
-                if i.as_string().is_err() {
-                    let mut tag = None;
-                    for arg in &self.args {
-                        if arg.item.contains("$it") {
-                            tag = Some(arg.tag());
-                        }
-                    }
-                    if let Some(tag) = tag {
-                        return Err(ShellError::labeled_error(
+            let input_strings = inputs.iter().map(|i| {
+                i.as_string().map_err(|_| {
+                    let arg = self.args.iter().find(|arg| arg.item.contains("$it"));
+                    if let Some(arg) = arg {
+                        ShellError::labeled_error(
                             "External $it needs string data",
                             "given row instead of string data",
-                            tag,
-                        ));
+                            arg.tag(),
+                        )
                     } else {
-                        return Err(ShellError::labeled_error(
+                        ShellError::labeled_error(
                             "Error: $it needs string data",
                             "given something else",
-                            name_tag,
-                        ));
+                            self.name_tag.clone(),
+                        )
                     }
-                }
-                if !first {
-                    process = process.arg("&&");
-                    process = process.arg(&self.name);
-                } else {
-                    first = false;
-                }
+                })
+            }).collect::<Result<Vec<String>, ShellError>>()?;
 
-                for arg in &self.args {
+            let commands = input_strings.iter().map(|i| {
+                let args = self.args.iter().filter_map(|arg| {
                     if arg.chars().all(|c| c.is_whitespace()) {
-                        continue;
+                        None
+                    } else {
+                        Some(arg.replace("$it", &i))
                     }
+                });
 
-                    process = process.arg(&arg.replace("$it", &i.as_string()?));
-                }
-            }
+                format!("{} {}", self.name, itertools::join(args, " "))
```

Comment:
This is perhaps the biggest point of discussion. Not sure if there's some escaping I'd have to do here to ensure the joined command below will be executed correctly by the shell.